app/models/candidate.rb

```diff
@@ -78,6 +78,16 @@ def pseudonymised_id
     Digest::SHA2.hexdigest(id.to_s)
   end
 
+  def delete
+    application_form_ids = application_forms.pluck(:id)
```

Comment:
Could we handle this with a `dependant: :destroy` on the `has_many: :application_work_experiences`?

Reply:
It doesn't seem to do the job `dependent: destroy or delete` I think it doesn't work because it's a polymorphic link